bokeh/core/property/tests/test_bases.py

```diff
@@ -65,3 +73,35 @@ def test_property_matches_incompatible_arrays(capsys):
     out, err = capsys.readouterr()
     # no way to suppress FutureWarning in this case
     # assert err == ""
+
+def test_property_matches_dicts_with_array_values(capsys):
+    p = pb.Property()
+    d1 = dict(foo=np.arange(10))
+    d2 = dict(foo=np.arange(10))
+
+    assert p.matches(d1, d1) is True
+    assert p.matches(d1, d2) is True
+
+    # XXX not sure if this is preferable to have match, or not
+    assert p.matches(d1, dict(foo=list(range(10)))) is True
```

Comment:
Right, that's the reason it returns `True` but I'm not sure that's what we'd actually want given that we do treat arrays and lists differently in some places